inc/global_assets/namespace.php

```diff
@@ -0,0 +1,343 @@
+		update_site_option( 'global_media_site', rtrim( $media_site_url, '/' ) );
+		update_site_option( 'global_media_site_id', $media_site_id );
+
+		// Configure WP AMF.
+		define( 'AMF_WORDPRESS_URL', $media_site_url );
+		return;
+	} catch ( Exception $error ) {
+		trigger_error( $error->getMessage(), E_USER_WARNING );
+
+		// Store the error.
+		update_site_option( 'global_media_site_error', $error->getMessage() );
+
+		// If we have no site and no URL remove the WP AMF plugin hooks.
+		remove_action( 'plugins_loaded', 'AMFWordPress\\register_settings' );
+		remove_action( 'admin_init', 'AMFWordPress\\register_settings_ui' );
+		remove_filter( 'amf/provider', 'AMFWordPress\\get_provider' );
+	}
+}
+
+/**
+ * Handle deletion of media site.
+ *
+ * @return void
+ */
+function uninitialize_media_site() {
+	if ( ! is_media_site() ) {
+		return;
+	}
+
+	delete_site_option( 'global_media_site' );
+	delete_site_option( 'global_media_site_id' );
+}
+
+/**
+ * Trim down the media site's admin menu.
+ *
+ * @return void
+ */
+function admin_menu() : void {
+	global $menu;
+
+	if ( ! is_media_site() ) {
+		return;
+	}
+
+	$allowed_menu_pages = [
+		'upload.php',
+		'users.php',
+	];
+
+	/**
+	 * Filters the admin menu pages allowed on the Global Media Site admin menu.
+	 *
+	 * @param array $allowed_menu_pages The page slugs allowed in the global media site admin menu.
+	 */
+	$allowed_menu_pages = (array) apply_filters( 'altis.media.global_site_menu_pages', [] );
+
+	// Always allow upload.php and users.php.
+	$allowed_menu_pages[] = 'upload.php';
+	$allowed_menu_pages[] = 'users.php';
+
+	foreach ( $menu as $position => $item ) {
+		if ( ! in_array( $item[2], $allowed_menu_pages, true ) ) {
+			unset( $menu[ $position ] );
+		}
+	}
+}
+
+/**
+ * Modify the admin bar for the media site.
+ *
+ * @param WP_Admin_Bar $wp_admin_bar The menu bar control.
+ * @return void
+ */
+function admin_bar_menu( WP_Admin_Bar $wp_admin_bar ) : void {
+
+	$wp_admin_bar->remove_node( sprintf( 'blog-%d-n', get_site_option( 'global_media_site_id' ) ) );
+	$wp_admin_bar->remove_node( sprintf( 'blog-%d-c', get_site_option( 'global_media_site_id' ) ) );
+	$wp_admin_bar->remove_node( sprintf( 'blog-%d-v', get_site_option( 'global_media_site_id' ) ) );
+
+	if ( ! is_media_site() ) {
+		return;
+	}
+
+	// Remove content and front end related menus.
+	$wp_admin_bar->remove_menu( 'new-content' );
+	$wp_admin_bar->remove_menu( 'comments' );
+	$wp_admin_bar->remove_menu( 'comments' );
+	$wp_admin_bar->remove_node( 'view-site' );
+
+	// Add the site title in as static text.
+	$wp_admin_bar->add_menu( [
+		'id' => 'site-name',
+		'title' => get_option( 'blogname' ),
+		'href' => false,
+	] );
+}
+
+/**
+ * Filter the site row actions in network admin to prevent deletion.
+ *
+ * @param array $actions The action links array.
+ * @param integer $site_id The site ID.
+ * @return array
+ */
+function sites_list_row_actions( array $actions, int $site_id ) : array {
```

Comment:
If we're removing all the quick actions, we may want to indicate that this site is set as the global media site in the site list, e.g.:

- [ ] | Site 1 &mdash; altis-dev.altis.dev/site-1/
- [ ] | Global Media Library &mdash; **Global Media Site** altis-dev.altis.dev/media/
- [ ] | Site Whatever &mdash; altis-dev.altis.dev/site-whatever/

It may be obvious with the site title, but if we're allowing the ability to define your own global media site (or if the site title is changed (which is still possible from the Edit Site screen), it may _not_ be that obvious.

Reply:
Yeah I saw the site states filter, can add that 👍